evennia/server/inputfuncs.py

```diff
@@ -654,6 +662,119 @@ def msdp_send(session, *args, **kwargs):
     session.msg(send=((), out))
 
 
+def is_typing_send_update():
+    """
+    Send relevant updates to participants
+
+    """
+    participants = list(_IS_TYPING_PARTICIPANTS.keys())
+
+    for participant in participants:
+        if _IS_TYPING_PARTICIPANTS[participant]["session"].puppet is not None:
+
+            payload = []
+            # Get potentials
+            potentials = (DefaultCharacter.objects.filter_family(db_location=_IS_TYPING_PARTICIPANTS[participant]["session"].puppet.location)
+                          .exclude(db_key=_IS_TYPING_PARTICIPANTS[participant]["session"].puppet.key))
```

Comment:
This could be a lot more readable if extracting `_IS_TYPING_PARTICIPANTS[participant]["session"] into a variable earlier. But then again, it seems much of this could be skipped if you track the participant object sessions directly.